File: ocaml/difference-of-squares/difference_of_squares.ml
open Base

let square_of_sum n =
  let sum = List.init n ~f:Int.succ |> List.fold ~init:0 ~f:(+)
  in sum * sum

let sum_of_squares n =
  List.init n ~f:(fun i -> (Int.succ i) * (Int.succ i))
  |> List.fold ~init:0 ~f:(+)

let difference_of_squares n = (square_of_sum n) - (sum_of_squares n)

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) (sum int) {
	for i :=0 ; i < n ; i++ {
		sum += (i + 1)
	}
	return sum * sum
}

func SumOfSquares(n int) (sum int) {
	for i :=0 ; i < n ; i++ {
		sum += ((i + 1) * (i + 1))
	}
	return
}

func Difference(n int) int { return (SquareOfSum(n) - SumOfSquares(n)) }
